src/pages/lib/auth/auth-events/q/platform/[platform].mdx

```diff
@@ -22,17 +22,298 @@ export const getStaticProps = (context) => {
+      break;
+    case 'forgotPasswordSubmit_failure':
+      console.log('password confirmation failed');
+      break;
+    case 'verify':
+      console.log('TOTP token verification successful');
+      break;
+    case 'tokenRefresh':
+      console.log('token refresh succeeded');
+      break;
+    case 'tokenRefresh_failure':
+      console.log('token refresh failed');
+      break;
+    case 'cognitoHostedUI':
+      console.log('Cognito Hosted UI sign in successful');
+      break;
+    case 'cognitoHostedUI_failure':
+      console.log('Cognito Hosted UI sign in failed');
+      break;
+    case 'customOAuthState':
+      console.log('custom state returned from CognitoHosted UI');
+      break;
+    case 'customState_failure':
+      console.log('custom state failure');
+      break;
+    case 'parsingCallbackUrl':
+      console.log('Cognito Hosted UI OAuth url parsing initiated');
+      break;
+    case 'userDeleted':
+      console.log('user deletion successful');
+      break;
+    case 'updateUserAttributes':
+      console.log('user attributes update successful');
+      break;
+    case 'updateUserAttributes_failure':
+      console.log('user attributes update failed');
+      break;
+    case 'signOut':
+      console.log('user signed out');
+      break;
+    default:
+      console.log('unknown event type');
+      break;
+  }
+};
+
+Hub.listen('auth', listener);
+```
+</Block>
+</BlockSwitcher>
+
+### Stop listening to events
+
+You can also stop listening for messages by calling the result of the `Hub.listen()` function. This may be useful if you no longer need to receive messages in your application flow. This can also help you avoid any memory leaks on low powered devices when you are sending large amounts of data through Amplify Hub on multiple channels.
+
+To stop listening to a certain event, you need to wrap the listener function with a variable and call it once you no longer need it:
+
+```javascript
+
+/* start listening for messages */
+const hubListenerCancelToken = Hub.listen('auth', (data) => {
+  console.log('Listening for all auth events: ', data.payload.data);
+});
+
+/* later */
+hubListenerCancelToken(); // stop listening for messages
+```
+
+You now have a few use cases and examples for listening to and responding to auth events.
+
+## Conclusion
 
-<Fragments fragments={{ js: js0 }} />
+Congratulations! You finished the **Listen to auth events** guide. In this guide, you learned how to listen for auth events across workflows.
 
-import ios1 from '/src/fragments/lib/auth/ios/hub_events/10_listen_events.mdx';
+## Next steps
+
+Now that you can listen to auth events you may also want to add some additional features. We recommend you learn more about:
+
+<InlineFilter filters={["js"]}>
+
+- [Manage user profile](/lib/auth/manageusers/q/platform/js/)
+- [Set up password change and recovery](/lib/auth/password_management/q/platform/js/)
+- [Use with Server-Side Rendering (SSR)](/lib/ssr/q/platform/js/)
```

Comment:
let us remove the SSR page from here - and then combine the links for the inline filters for both JS + RN

Reply:
@abdallahshaban557 Removed SSR but the links are set up so they only see the links relevant to their platform. In the new site structure we can take these away but for now the platform switcher does not allow for relative links